---
title: "RNAseq Analysis Example"
output: html_document
---
This is an introduction to RNAseq analysis for use at Software Carpentry bootcamps that have covered novice R. It involves reading in some count data from an RNAseq experiment, exploring the data and then analysis with the package DESeq2. 

```{r data_input}
## Filename with output from featureCounts
countfile <- "data/counts.txt"
## Read in the data
countdata <- read.table(countfile, header=TRUE, row.names=1)
head(countdata)
colnames(countdata)
```
Have a look at the data. It contains genes (rows) with information about the gene positions in the first five columns and then information about the number of reads aligning to the gene in each experimental sample. We don't need the information on gene position, so we can remove it from the data frame.

```{r}

# Remove first five columns (chr, start, end, strand, length)
countdata <- countdata[ ,-(1:5)] 
head(countdata)
colnames(countdata)
```